### DISCUSSION TITLE
What is the recommened way to compile for windows?

### DISCUSSION THREAD

**Discussion by Volumie** [LMMS/lmms#6024] [General]
**I was wondering how the binaries for windows were built by @Spekular, and what is the recommended way to do it?**

Can someone suggest me the best option, so that all the features will be available in the build?
Is it:
1. Compile on windows using msys.
2. Compile on windows using msvc.
3. Cross-compile on ubuntu for windows.

**Comment by Spekular** [comment]
I've never built for Windows. I suppose you're looking at the fact that I tagged the latest release, but @PhysSong handled the actual build part.

**Comment by Volumie** [comment]
@PhysSong Can you please tell me how you create binaries for windows?

**Comment by PhysSong** [comment]
The 3rd way is what we use for nightly builds. You can use `lmmsci/linux.mingw64:18.04` or `lmmsci/linux.mingw32:18.04` Docker images on any Linux distributions as well.

**Comment by mikemartell** [comment]
In case you're still interested, I just posted my notes for building on Windows : https://pastebin.com/Q5ULqg6y
Note: Until I can get a PR approved, you'll need to manually copy the CMakeCache.txt to the runtime directory.

**Comment by anytizer** [comment]
Some notes on how I managed to [Compile under windows using Visual Studio](https://gist.github.com/anytizer/a674f2d24cd91dd2ffc510963e1a241c) - second option from above.

**Comment by tresf** [comment]
We (finally) have MSVC documented here: https://github.com/LMMS/lmms/wiki/dependencies-windows#windows-msvc.  If this needs updating, please ping a dev (here, Discord) or obtain access to edit it yourself. <3
